### PR TITLE
Fix a panic when creating HVS app or secret

### PR DESCRIPTION
**Pull request by divyapola5**
### :hammer_and_wrench: Description
This change adds in a fix for the [bug](https://hashicorp.atlassian.net/browse/HV-3233). There is a panic when creating an app or a secret and this change addresses the issue.

### :building_construction: Acceptance tests

- [ ] Are there any feature flags that are required to use this functionality?
- [ ] Have you added an acceptance test for the functionality being added?
- [x] Have you run the acceptance tests on this branch?

Output from acceptance testing:
Screenshot 2024-12-13 at 3 11 05 PM
Screenshot 2024-12-13 at 3 11 24 PM
Screenshot 2024-12-13 at 3 25 21 PM

```sh
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: .changelog/1154.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+creating a hvs app or kv secret should not panic
```
Comment (rodrigo-hcp) [inline .changelog/1154.txt:2]: _nit_: I'd rather just repeat the PR title here

```suggestion
Fix a panic when creating HVS app or secret
```

Comment (divyapola5) [reply .changelog/1154.txt]: done

**Review by rodrigo-hcp — APPROVED** [review]

**Comment by maxcoulombe** [comment]
The changes look good. I did find another bug that we could fix at the same time or as a follow-up PR - up to you.

If you change the name of a hcp_vault_secrets_secret I'd expect that the secret with the previous name gets deleted and the a secret with the new name gets created. However, the TF provider treats it as an update and creates the new secret while not deleting the previous one. Same if the app name changes, the resource gets duplicated instead of recreated.

I think it can be simply fixed by adding this to the "app_name" and "secret_name" on the hcp_vault_secrets_secret schema.
```
PlanModifiers: []planmodifier.String{
stringplanmodifier.RequiresReplace(),
},
```

Does it make sense?

**Review by maxcoulombe — APPROVED** [review]

**Comment by divyapola5** [comment]
> The changes look good. I was able to reproduce the panic then when building this branch it fixed my TF project.
> 
> I did find another bug that we could fix at the same time or as a follow-up PR - up to you.
> 
> If you change the name of a hcp_vault_secrets_secret I'd expect that the secret with the previous name gets deleted and the a secret with the new name gets created. However, the TF provider treats it as an update and creates the new secret while not deleting the previous one. Same if the app name changes, the resource gets duplicated instead of recreated.
> 
> I think it can be simply fixed by adding this to the "app_name" and "secret_name" on the hcp_vault_secrets_secret schema.
> 
> ```
> PlanModifiers: []planmodifier.String{
> stringplanmodifier.RequiresReplace(),
> },
> ```
> 
> Does it make sense?

Ah did not know about this. Sure, will add it as part of this PR.

**Review by jasonpilz — APPROVED** [review]

**Pull request merged**